Swift/AEPSampleApp/EdgeIdentityView.swift

```diff
@@ -7,13 +7,95 @@
  it.
  */
 
-import Foundation
+import AdSupport
+import AEPCore
 import AEPEdgeIdentity
+import AppTrackingTransparency
+import Foundation
 import SwiftUI
 
 struct EdgeIdentityView: View {
     @State var currentEcid = ""
     @State var currentIdentityMap: IdentityMap?
+    @State var adID: UUID?
+    @State var adIdText: String = ""
+    @State var trackingAuthorizationResultText: String = ""
+    
+    /// Provides the `advertisingIdentifier` for the given environment, assuming tracking authorization is provided
+    ///
+    /// Simulators will never provide a valid UUID, regardless of authorization; use the set ad ID flow to test a specific ad ID.
+    func getAdvertisingIdentifierForEnvironment() -> UUID {
+        #if targetEnvironment(simulator)
+        print("""
+            Simulator environment detected. Please note that simulators cannot retrieve valid advertising identifier
+            from the ASIdentifierManager (as specified by Apple); an all-zeros UUID will be retrieved even if
+            authorization is provided. If you want to use a specific ad ID, you can use the set ad ID flow.
+            """)
+        #endif
+        print("Advertising identifier: \(ASIdentifierManager.shared().advertisingIdentifier)")
+        return ASIdentifierManager.shared().advertisingIdentifier
+    }
+    
+    /// Requests tracking authorization from the user; prompt will only be shown once per app install, as per Apple rules
+    ///
+    /// It is possible to change tracking permissions at the Settings app level. Any change in tracking permissions will terminate the app.
+    /// It is also possible for system-wide tracking to be off but individual per-app permissions granted.
+    /// If "Allow Apps to Request to Track" at the system level was on and is turned off, a system prompt appears asking if previously provided individual per-app tracking permissions should be kept as-is or all turned off
+    func requestTrackingAuthorization() {
```

Comment:
Thanks for setting this up @timkimadobe!
As we discussed we can keep the generic code for requesting authorization and initial setting of adid in the AppDelegate, and any helper buttons or UI elements under the Edge Identity view.